src/Microsoft.Data/DataFrame.cs

```diff
@@ -34,6 +35,132 @@ public DataFrame(IList<BaseColumn> columns)
             _table = new DataFrameTable(columns);
         }
 
+        public DataFrame(RecordBatch recordBatch)
+        {
+            _table = new DataFrameTable();
+            Apache.Arrow.Schema arrowSchema = recordBatch.Schema;
+            int fieldIndex = 0;
+            IEnumerable<IArrowArray> arrowArrays = recordBatch.Arrays;
+            foreach (IArrowArray arrowArray in arrowArrays)
+            {
+                Field field = arrowSchema.GetFieldByIndex(fieldIndex);
+                IArrowType fieldType = field.DataType;
+                BaseColumn dataFrameColumn = null;
+                switch (fieldType.TypeId)
+                {
+                    case ArrowTypeId.Boolean:
+                        ReadOnlyMemory<byte> valueBuffer = (arrowArray as BooleanArray).ValueBuffer.Memory;
```

Comment:
Instead of a pattern like:

`(someObj as SomeType).SomeProperty`

in general it is better to cast:

`((SomeType)someObj).SomeProperty`

The reasoning is because if the cast doesn't succeed in the first case, you get a NullReferenceException. In the second case, you get a InvalidCastException. It is easier to diagnose the error using the second pattern.